synth.py

```diff
@@ -27,28 +32,82 @@
 gapic = gcp.GAPICMicrogenerator()
 common_templates = gcp.CommonTemplates()
 
-versions = ['v1', 'v1beta1']
+versions = ["v1", "v1beta1"]
 
 for version in versions:
     library = gapic.typescript_library(
-        'texttospeech',
+        "texttospeech",
         generator_args={
             "grpc-service-config": f"google/cloud/texttospeech/{version}/texttospeech_grpc_service_config.json",
-            "package-name":f"@google-cloud/text-to-speech"
-            },
-        proto_path=f'/google/cloud/texttospeech/{version}',
-        version=version)
+            "package-name": f"@google-cloud/text-to-speech",
+        },
+        proto_path=f"/google/cloud/texttospeech/{version}",
+        version=version,
+    )
 
     # skip index, protos, package.json, and README.md
     s.copy(
-        library,
-        excludes=['package.json', 'README.md', 'src/index.ts'],
+        library, excludes=["package.json", "README.md", "src/index.ts"],
     )
 
-templates = common_templates.node_library(source_location='build/src')
+
+def find_clients(filePath: str) -> typing.List[str]:
+    with open(filePath, "r") as fh:
+        content = fh.read()
+    return re.findall(r"\{(.*Client)\}", content)
+
+
+def process_template(versions: typing.List[str], default_version: str, clients: typing.List[str]) -> None:
+    templateLoader = FileSystemLoader(searchpath="./templates")
+    templateEnv = Environment(loader=templateLoader)
+    TEMPLATE_FILE = "index.ts.jinja2"
+    index = templateEnv.get_template(TEMPLATE_FILE)
+    # this is where to put args to the template renderer
+    outputText = index.render(
+        versions=versions, default_version=default_version, clients=clients)
+    with open("src/index.ts", "w") as fh:
+        fh.write(outputText)
+
+
+def generate_index_ts(versions: typing.List[str], default_version: str) -> None:
+    """Parses src/*/index.ts...."""
+    versioned_index_ts_path = pathlib.Path(
+        "src") / default_version / "index.ts"
+    # TODO(Summer): replace the tsc command with a command that lists the exports.
+    # output: str = subprocess.run(
+    #     ["tsc", "--arg1", "yadayada", str(versioned_index_ts_path)],
```

Comment:
I don't think there is a good way to extact list of exports using `tsc`. I'm OK with using regex here.